atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -165,6 +167,39 @@ protected boolean shouldStopRetrying(int numTimesFailed) {
     public TransactionAndImmutableTsLock setupRunTaskWithConditionThrowOnConflict(PreCommitCondition condition) {
         StartIdentifiedAtlasDbTransactionResponse transactionResponse
                 = timelockService.startIdentifiedAtlasDbTransaction();
+        return wrap(condition, transactionResponse);
+    }
+
+    @Override
+    public List<Optional<TransactionAndImmutableTsLock>> setupRunTaskWithConditionThrowOnConflictBatch(
+            List<PreCommitCondition> conditions) {
+        List<Optional<StartIdentifiedAtlasDbTransactionResponse>> transactionResponses
+                = timelockService.startIdentifiedAtlasDbTransactionBatch(conditions.size());
+
+        if (transactionResponses.stream().allMatch(response -> !response.isPresent())) {
+            // (todo) jshah - We may or may not want to throw in the case that all transactions failed to start;
+            //      to be decided
+            throw new RuntimeException("entire batch failed");
```

Comment:
It's pointless to have this in the API level down if you're just going to throw.